src/Agent.Listener/Configuration/RSACngKeyManager.cs

```diff
@@ -0,0 +1,108 @@
+﻿#if OS_WINDOWS
```

Comment:
I don't think we should include classes that don't apply to the target platform. In this case the Windows implementation is able to use OS features to enforce security constraints on the keys, while on non-windows platforms we use file based security using chmod. I don't disagree with your statements, but since we build on all platforms anyway I'm not sure I agree that both implementations should be included for all OSes.